src/snowflake/snowpark/async_job.py

```diff
@@ -331,6 +333,15 @@ def result(
             _AsyncResultType(result_type.lower()) if result_type else self._result_type
         )
         self._cursor.get_results_from_sfqid(self.query_id)
+        if self._num_statements is not None:
+            for _ in range(self._num_statements - 1):
+                self._cursor.nextset()
+            # TODO: Once we support fetch_pandas_all for multi-statement query in connector, we could remove this
+            #   workaround.
```

Comment:
Should we have a JIRA for this

Reply:
Created [SNOW-840818](https://snowflakecomputing.atlassian.net/browse/SNOW-840818).